policy/pa.go

```diff
@@ -174,24 +174,24 @@ func isDNSCharacter(ch byte) bool {
 
 var (
 	errInvalidIdentifier    = berrors.MalformedError("Invalid identifier type")
-	errNonPublic            = berrors.MalformedError("Name does not end in a public suffix")
-	errICANNTLD             = berrors.MalformedError("Name is an ICANN TLD")
-	errPolicyForbidden      = berrors.RejectedIdentifierError("Policy forbids issuing for name")
-	errInvalidDNSCharacter  = berrors.MalformedError("Invalid character in DNS name")
-	errNameTooLong          = berrors.MalformedError("DNS name too long")
-	errIPAddress            = berrors.MalformedError("Issuance for IP addresses not supported")
-	errTooManyLabels        = berrors.MalformedError("DNS name has too many labels")
-	errEmptyName            = berrors.MalformedError("DNS name was empty")
-	errNameEndsInDot        = berrors.MalformedError("DNS name ends in a period")
-	errTooFewLabels         = berrors.MalformedError("DNS name does not have enough labels")
-	errLabelTooShort        = berrors.MalformedError("DNS label is too short")
-	errLabelTooLong         = berrors.MalformedError("DNS label is too long")
-	errMalformedIDN         = berrors.MalformedError("DNS label contains malformed punycode")
-	errInvalidRLDH          = berrors.RejectedIdentifierError("DNS name contains a R-LDH label")
-	errTooManyWildcards     = berrors.MalformedError("DNS name had more than one wildcard")
-	errMalformedWildcard    = berrors.MalformedError("DNS name had a malformed wildcard label")
-	errICANNTLDWildcard     = berrors.MalformedError("DNS name was a wildcard for an ICANN TLD")
-	errWildcardNotSupported = berrors.MalformedError("Wildcard names not supported")
+	errNonPublic            = berrors.MalformedError("Domain name does not end with a valid public suffix (TLD)")
+	errICANNTLD             = berrors.MalformedError("Domain name ends with a reserved ICANN TLD. Certificate can not be issued")
+	errPolicyForbidden      = berrors.RejectedIdentifierError("Policy forbids issuing a certificate for the domain name. Part or all of the name may be blacklisted")
+	errInvalidDNSCharacter  = berrors.MalformedError("Domain name contains invalid character")
+	errNameTooLong          = berrors.MalformedError("Domain name too long. (Note that certificates containing domain names over 230 octets long are not currently supported)")
+	errIPAddress            = berrors.MalformedError("Certificate issue for IP addresses is not supported")
```

Comment:
```suggestion
	errIPAddress            = berrors.MalformedError("Certificate issuance for IP addresses is not supported")
```

Reply:
Fair. But does anyone say "issuance" not "issue" in everyday language? Is there any actual reason not to go with the equally accurate but much more commonplace word? (Do we say "obviated" rather than "not needed/not necessary"? It comes over a bit like that. But its also minor, happy to leave if you think otherwise)

Reply:
"issue" as a noun means something different than "issuance" as a noun. It would mean "problem" or similar. If we treat "issue" as a verb in your proposal it would be ungrammatical. We could try:

"The ACME server will not issue certificates for IP addresses"

This has the added advantage of removing the passive voice.

Reply:
Agree. I think this is also US/UK English divergence to an extent. Yours is better